schema/shared/src/test/scala/zio/blocks/schema/DynamicPatchSpec.scala

```diff
@@ -0,0 +1,398 @@
+        val ops = StringOp.diff("hello", "hlo")
+        assertTrue(ops.nonEmpty)
+      }
+    ),
+    suite("SeqOp.diff")(
+      test("diff identical sequences produces empty ops") {
+        val a = Vector(
+          DynamicValue.Primitive(PrimitiveValue.Int(1)),
+          DynamicValue.Primitive(PrimitiveValue.Int(2))
+        )
+        val ops = SeqOp.diff(a, a)
+        assertTrue(ops.isEmpty || ops.forall {
+          case SeqOp.Modify(_, _) => true
+          case _ => false
+        })
+      },
+      test("diff computes insert for added elements") {
+        val a = Vector(DynamicValue.Primitive(PrimitiveValue.Int(1)))
+        val b = Vector(
+          DynamicValue.Primitive(PrimitiveValue.Int(1)),
+          DynamicValue.Primitive(PrimitiveValue.Int(2))
+        )
+        val ops = SeqOp.diff(a, b)
+        assertTrue(ops.exists {
+          case SeqOp.Insert(_, _) | SeqOp.Append(_) => true
+          case _ => false
+        })
+      },
+      test("diff computes delete for removed elements") {
+        val a = Vector(
+          DynamicValue.Primitive(PrimitiveValue.Int(1)),
+          DynamicValue.Primitive(PrimitiveValue.Int(2))
+        )
+        val b = Vector(DynamicValue.Primitive(PrimitiveValue.Int(1)))
+        val ops = SeqOp.diff(a, b)
+        assertTrue(ops.exists {
+          case SeqOp.Delete(_, _) => true
+          case _ => false
+        })
+      }
+    ),
+    suite("MapOp.diff")(
+      test("diff identical maps produces empty ops") {
+        val entries = Vector(
+          DynamicValue.Primitive(PrimitiveValue.String("a")) -> DynamicValue.Primitive(PrimitiveValue.Int(1))
+        )
+        val ops = MapOp.diff(entries, entries)
+        assertTrue(ops.isEmpty)
+      },
+      test("diff computes add for new keys") {
+        val a = Vector(
+          DynamicValue.Primitive(PrimitiveValue.String("a")) -> DynamicValue.Primitive(PrimitiveValue.Int(1))
+        )
+        val b = Vector(
+          DynamicValue.Primitive(PrimitiveValue.String("a")) -> DynamicValue.Primitive(PrimitiveValue.Int(1)),
+          DynamicValue.Primitive(PrimitiveValue.String("b")) -> DynamicValue.Primitive(PrimitiveValue.Int(2))
+        )
+        val ops = MapOp.diff(a, b)
+        assertTrue(ops.exists {
+          case MapOp.Add(_, _) => true
+          case _ => false
+        })
+      },
+      test("diff computes remove for deleted keys") {
+        val a = Vector(
+          DynamicValue.Primitive(PrimitiveValue.String("a")) -> DynamicValue.Primitive(PrimitiveValue.Int(1)),
+          DynamicValue.Primitive(PrimitiveValue.String("b")) -> DynamicValue.Primitive(PrimitiveValue.Int(2))
+        )
+        val b = Vector(
+          DynamicValue.Primitive(PrimitiveValue.String("a")) -> DynamicValue.Primitive(PrimitiveValue.Int(1))
+        )
+        val ops = MapOp.diff(a, b)
+        assertTrue(ops.exists {
+          case MapOp.Remove(_) => true
+          case _ => false
+        })
+      }
+    ),
```

Comment:
I think you need property-based tests, because we have laws we need to guarantee, e.g. (`a.diff(b).apply(a) == b`, for all `a` and `b`).